### PR TITLE
Add Ktlint reformatting as an IntelliJ on save action

### PR DESCRIPTION
**Pull request by tlusk**
This implements the ActionOnSave extension causing Ktlint reformatting to show up as an option under the "Actions on Save" settings page.

This resolves issue #125

### CODE REVIEW

**Review by nbadal — COMMENTED** [review]
Great addition to the plugin, thank you!

File: plugin/src/main/kotlin/com/nbadal/ktlint/KtlintActionOnSaveInfoProvider.kt [hunk 1]
```diff
@@ -0,0 +1,50 @@
+package com.nbadal.ktlint
+
+import com.intellij.ide.actionsOnSave.*
+
+class KtlintActionOnSaveInfoProvider : ActionOnSaveInfoProvider() {
+    override fun getActionOnSaveInfos(context: ActionOnSaveContext): List<ActionOnSaveInfo> = listOf(PrettierOnSaveActionInfo(context))
+
+    override fun getSearchableOptions(): Collection<String> {
+        return listOf("ktlint")
```
[the remark this answers is not in the document]
Comment (nbadal) [reply plugin/src/main/kotlin/com/nbadal/ktlint/KtlintActionOnSaveInfoProvider.kt:9]: Though I'm not actually sure where the Searchable is used 🤔 if you know, please let me know!

File: plugin/src/main/kotlin/com/nbadal/ktlint/KtlintActionOnSaveInfoProvider.kt [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    }
+}
+
+private const val KTLINT_CONFIGURABLE_ID = "preferences.ktlint"
+
+private class PrettierOnSaveActionInfo(actionOnSaveContext: ActionOnSaveContext) :
```
[the remark this answers is not in the document]
Comment (tlusk) [reply plugin/src/main/kotlin/com/nbadal/ktlint/KtlintActionOnSaveInfoProvider.kt:15]: Haha oops I missed that one

**Comment by tlusk** [comment]
@nbadal the above comments have been addressed

File: plugin/src/main/kotlin/com/nbadal/ktlint/KtlintActionOnSaveInfoProvider.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (tlusk) [reply plugin/src/main/kotlin/com/nbadal/ktlint/KtlintActionOnSaveInfoProvider.kt]: It's used by the search box in the settings.  The documentation for the interface also says returning the save action name is a good idea so I did change it.

> Strings that help to find 'Actions on Save' page when using 'Find Action' or 'Search 
> Everywhere' or when using a search field in Settings (Preferences). Typically, 
> implementations return a singleton list with ActionOnSaveInfo.getActionOnSaveName().

**Comment by nbadal** [comment]
Looks great @tlusk thank you for the contribution!

**Pull request merged**